src/utils/hooks/useGetSectionProgress.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState, useEffect } from 'react'
+import { getFirstErrorLocation, revalidateAll } from '../helpers/application/revalidateAll'
+import {
+  ApplicationElementStates,
+  ResponsesByCode,
+  SectionsProgress,
+  TemplateSectionPayload,
+  User,
+} from '../types'
+
+interface UseGetSectionProgressProps {
+  loadStart: boolean
+  currentUser: User
+  serialNumber: string
+  templateSections: TemplateSectionPayload[]
+  elementsState: ApplicationElementStates
+  responsesByCode: ResponsesByCode
+}
+
+const useGetSectionsProgress = ({
+  loadStart,
+  currentUser,
+  serialNumber,
+  templateSections,
+  elementsState,
+  responsesByCode,
+}: UseGetSectionProgressProps) => {
+  const [sections, setSections] = useState<SectionsProgress>()
+  const [isLoadingProgress, setIsLoadingProgress] = useState(false)
+
+  useEffect(() => {
+    if (loadStart && currentUser && templateSections && elementsState && responsesByCode) {
+      setIsLoadingProgress(true)
+      runScanSectionProgress()
+    }
+  }, [loadStart])
+
+  async function runScanSectionProgress() {
+    let sectionsProgress: SectionsProgress = {}
+    templateSections.forEach(async ({ code, title, index }) => {
+      const validate = await revalidateAll({
+        elementsState,
+        responsesByCode,
+        currentUser,
+        sectionCode: code,
+        strict: false, // Do we still need this flag?
+      })
+
+      const { progress, validityFailures } = validate
+      const { total, done, valid } = progress
+
+      sectionsProgress = {
+        ...sectionsProgress,
+        [index]: {
+          info: {
+            title,
+            code,
+          },
+          progress: { total, done, valid, completed: valid && total === done },
+          link: !(total === done && valid) ? getLinkToSection(validityFailures) : undefined,
+        },
+      }
+
+      setSections(sectionsProgress)
+      if (templateSections.length === Object.keys(sectionsProgress).length)
+        setIsLoadingProgress(false)
+    })
+  }
+
+  const getLinkToSection = (validityFailures: any): string => {
```

Comment:
Am I correct in my understanding that this will get a link to the Section AND Page? If so, maybe call it `getLinkToLocation` or `getLinkToInProgressLocation` ?